### PR TITLE
Support edma3 for s32z270dc2

### PR DESCRIPTION
**Pull request by haduongquang**
Add EDMAv3 instances base addresses and module features to support using the EDMA MCUX driver.

### CODE REVIEW

File: s32/mcux/devices/S32Z270/drivers/fsl_memory.h
Comment (manuargue) [inline s32/mcux/devices/S32Z270/drivers/fsl_memory.h:1]: why do we have more functionalities implemented in `s32/mcux/devices/S32K344/drivers/fsl_memory.h` than on this file? aren't they required?

Comment (Dat-NguyenDuy) [reply s32/mcux/devices/S32Z270/drivers/fsl_memory.h:1]: Not in this case. For S32K3xx, the DMA accesses TCM memory over "backdoor", the core can access directly.
So if TCM is used for DMA, the DMA driver will need `s32/mcux/devices/S32K344/drivers/fsl_memory.h` to convert from address space being used by the core to address space being used by DMA

File: s32/mcux/devices/S32Z270/S32Z270_glue_mcux.h [hunk 1]
```diff
@@ -247,4 +247,49 @@
...
+/** Peripheral EDMA3_5 base pointer */
+#define EDMA3_5                                     ((DMA_Type *)EDMA3_5_BASE)
+/** Interrupt vectors for the DMA3_5 peripheral type */
+#define DMA3_5_IRQS                                 { RTU_DMA5_0_IRQn, RTU_DMA5_1_IRQn, RTU_DMA5_2_IRQn, RTU_DMA5_3_IRQn, RTU_DMA5_4_IRQn, RTU_DMA5_5_IRQn, RTU_DMA5_6_IRQn, RTU_DMA5_7_IRQn, RTU_DMA5_8_IRQn, RTU_DMA5_9_IRQn, RTU_DMA5_10_IRQn, RTU_DMA5_11_IRQn, RTU_DMA5_12_IRQn, RTU_DMA5_13_IRQn, RTU_DMA5_14_IRQn, RTU_DMA5_15_IRQn }
+#define DMA3_5_ERROR_IRQS                           { RTU_DMA5_ERR_IRQn }
+
+/** Peripheral DMA3 base address */
+#define DMA3_BASE                                   {EDMA3_0_BASE, EDMA3_1_BASE, EDMA3_4_BASE, EDMA3_5_BASE}
+/** Peripheral DMA3 base pointer */
+#define DMA3                                        {EDMA3_0, EDMA3_1, EDMA3_4, EDMA3_5}
```
Comment (Dat-NguyenDuy) [inline s32/mcux/devices/S32Z270/S32Z270_glue_mcux.h:286]: Hmm, i'm getting lost here what you define, is the `DMA3` here same as `DMA_BASE_PTRS`; `DMA3_BASE` and `DMA_BASE_ADDRS`. Also i see there is expression like:
`#define FSL_FEATURE_EDMA_INSTANCE_HAS_TRANSFER_MODEn(x) (((x) == DMA3) ? (0) : (-1))`
It is seems an invalid expression (comparing a number with an array?). Pls refine the definition in here

**Review by Dat-NguyenDuy — CHANGES_REQUESTED** [review]

File: s32/mcux/devices/S32Z270/S32Z270_glue_mcux.h (hunk 1 above)
Comment (haduongquang) [reply s32/mcux/devices/S32Z270/S32Z270_glue_mcux.h]: For the first point, DMA3 and DMA3_BASE are the same with DMA_BASE_PTRS and DMA_BASE_ADDRS, so I removed it.
And the second point, I updated the definition.

**Review by Dat-NguyenDuy — APPROVED** [review]

File: s32/mcux/devices/S32Z270/S32Z270_features.h [hunk 2]
```diff
@@ -110,6 +110,135 @@
...
+#define FSL_FEATURE_SOC_EDMA_COUNT (5)
+
+/* EDMA module features */
+
+/* @brief Number of DMA channels (related to number of registers TCD, DCHPRI, bit fields ERQ[ERQn], EEI[EEIn], INT[INTn], ERR[ERRn], HRS[HRSn] and bit field widths ES[ERRCHN], CEEI[CEEI], SEEI[SEEI], CERQ[CERQ], SERQ[SERQ], CDNE[CDNE], SSRT[SSRT], CERR[CERR], CINT[CINT], TCDn_CITER_ELINKYES[LINKCH], TCDn_CSR[MAJORLINKCH], TCDn_BITER_ELINKYES[LINKCH]). (Valid only for eDMA modules.) */
+#define FSL_FEATURE_EDMA_MODULE_CHANNEL (32)
+/* @brief If 8 bytes transfer supported. */
+#define FSL_FEATURE_EDMA_SUPPORT_8_BYTES_TRANSFER (1)
+/* @brief Number of channel for each EDMA instance, (only defined for soc with different channel numbers for difference instance) */
+#define FSL_FEATURE_EDMA_INSTANCE_CHANNELn(x) (32)
```
Comment (manuargue) [inline s32/mcux/devices/S32Z270/S32Z270_features.h:123]: how this relates to https://github.com/zephyrproject-rtos/zephyr/pull/78368/files#diff-9ae4a85169b9753909d6b33ec1185b4e689e9d1a76182de5458cd9b9a4601a86R171-R172 ?

File: s32/mcux/devices/S32Z270/S32Z270_features.h [hunk 3]
```diff
@@ -110,6 +110,135 @@
...
+/* @brief If 128 bytes transfer supported. */
+#define FSL_FEATURE_EDMA_INSTANCE_SUPPORT_128_BYTES_TRANSFERn(x)  \
+    (((x) == EDMA3_0) ? (0) : \
+    (((x) == EDMA3_1) ? (0) : \
+    (((x) == EDMA3_4) ? (0) : \
+    (((x) == EDMA3_5) ? (0) : (-1)))))
+/* @brief If channel clock controlled independently */
+#define FSL_FEATURE_EDMA_CHANNEL_HAS_OWN_CLOCK_GATE (1)
+/* @brief NBYTES must be multiple of 8 when using scatter gather. */
+#define FSL_FEATURE_EDMA_INSTANCE_HAS_ERRATA_51327n(x)  \
```
Comment (manuargue) [inline s32/mcux/devices/S32Z270/S32Z270_features.h:151]: why not just 0 like `#define FSL_FEATURE_EDMA_HAS_ERRATA_51327 (0)`? same for any other feature that always expand to 0 for all instances

**Review by manuargue — CHANGES_REQUESTED** [review]
block until zephyr pr approved

File: s32/mcux/devices/S32Z270/S32Z270_features.h (hunk 2 above)
Comment (haduongquang) [reply s32/mcux/devices/S32Z270/S32Z270_features.h]: Yes, I have updated this definition, instance 0 (32) and other (16).

File: s32/mcux/devices/S32Z270/S32Z270_features.h (hunk 3 above)
Comment (haduongquang) [reply s32/mcux/devices/S32Z270/S32Z270_features.h]: I updated

File: s32/mcux/devices/S32Z270/drivers/fsl_memory.h
Comment (haduongquang) [reply s32/mcux/devices/S32Z270/drivers/fsl_memory.h:1]: Hi, I have removed `s32/mcux/devices/S32K344/drivers/fsl_memory.h` file that it's not required.

File: s32/mcux/devices/S32Z270/S32Z270_glue_mcux.h [hunk 4]
```diff
@@ -247,4 +247,45 @@
...
+/** Peripheral EDMA3_4 base pointer */
+#define EDMA3_4                                     ((DMA_Type *)EDMA3_4_BASE)
+/** Interrupt vectors for the DMA3_4 peripheral type */
+#define DMA3_4_IRQS                                 { RTU_DMA4_0_IRQn, RTU_DMA4_1_IRQn, RTU_DMA4_2_IRQn, RTU_DMA4_3_IRQn, RTU_DMA4_4_IRQn, RTU_DMA4_5_IRQn, RTU_DMA4_6_IRQn, RTU_DMA4_7_IRQn, RTU_DMA4_8_IRQn, RTU_DMA4_9_IRQn, RTU_DMA4_10_IRQn, RTU_DMA4_11_IRQn, RTU_DMA4_12_IRQn, RTU_DMA4_13_IRQn, RTU_DMA4_14_IRQn, RTU_DMA4_15_IRQn }
+#define DMA3_4_ERROR_IRQS                           { RTU_DMA4_ERR_IRQn }
+
+/** Peripheral EDMA3_5 base address */
+#define EDMA3_5_BASE                                IP_EDMA_5_MP_BASE
+/** Peripheral EDMA3_5 base pointer */
+#define EDMA3_5                                     ((DMA_Type *)EDMA3_5_BASE)
```
Comment (manuargue) [inline s32/mcux/devices/S32Z270/S32Z270_glue_mcux.h:278]: why you are defining macros named EDMA3_, DMA3_, DMA_? what's with some many prefixes? please defined only what is needed following the template of mcux-sdk device definitions

**Review by manuargue — CHANGES_REQUESTED** [review]

File: s32/mcux/devices/S32Z270/S32Z270_glue_mcux.h (hunk 4 above)
Comment (haduongquang) [reply s32/mcux/devices/S32Z270/S32Z270_glue_mcux.h]: I redefined the macro name to `DMA_` and removed the unnecessary definitions.

**Review by manuargue — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by manuargue** [comment]
zephyr pr approved by assignee, merging this

**Pull request merged**